tlspuffin/src/wolfssl/mod.rs

```diff
@@ -71,12 +72,45 @@ pub fn new_wolfssl_factory() -> Box<dyn Factory<TLSProtocolBehavior>> {
             Ok(Box::new(WolfSSL::new(config)?))
         }
 
+        fn kind(&self) -> PutKind {
+            PutKind::Rust
+        }
+
         fn name(&self) -> PutName {
             WOLFSSL520_PUT
         }
 
-        fn version(&self) -> String {
-            WolfSSL::version()
+        fn versions(&self) -> Vec<(String, String)> {
+            let wolfssl_shortname = if cfg!(feature = "vendored-wolfssl540") {
+                "wolfssl 5.4.0-stable"
+            } else if cfg!(feature = "vendored-wolfssl530") {
+                "wolfssl 5.3.0-stable"
+            } else if cfg!(feature = "vendored-wolfssl520") {
+                "wolfssl 5.2.0-stable"
+            } else if cfg!(feature = "vendored-wolfssl510") {
+                "wolfssl 5.1.0-stable"
+            } else if cfg!(feature = "vendored-wolfssl430") {
+                "wolfssl 4.3.0-stable"
+            } else if cfg!(feature = "vendored-master") {
+                "master"
+            } else {
+                "unknown"
+            };
```

Comment:
Same here, I think it would be nice to get the real version that is executed. This could also be nice for verifying a build.

Reply:
The shortname is not a substitute for the version. The full version is also provided below, and I adapted the CLI to log both. Here is an example output:
```
2024-03-28T11:06:50+01:00       Put Versions:
2024-03-28T11:06:50+01:00       (Rust) rust-tcp:
2024-03-28T11:06:50+01:00           harness: TCP_______ (tlspuffin 0.1.0+git.1ec568)
2024-03-28T11:06:50+01:00       (Rust) rust-openssl:
2024-03-28T11:06:50+01:00           harness: OPENSSL111 (tlspuffin 0.1.0+git.1ec568)
2024-03-28T11:06:50+01:00           library: openssl (openssl111k / OpenSSL 1.1.1k  25 Mar 2021)
```